portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,130 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private List<String> countries;
```

Comment:
best practice: make these fields `final` and initialize them at declaration (`new ArrayList()`);

`final` fields are always better than mutable ones because it's impossible to have initialization bugs :-) it makes it much easier to reason about the code (e.g. is it possible for some code path to access this field prior to init()) and more maintainable.